ethcore/src/client/client.rs

```diff
@@ -297,27 +296,27 @@ impl Importer {
 			trace_time!("import_verified_blocks");
 			let start = Instant::now();
 
-			for block in blocks {
-				let header = block.header.clone();
-				let bytes = block.bytes.clone();
-				let hash = header.hash();
+			for mut block in blocks {
+				let hash = block.header.hash();
 
-				let is_invalid = invalid_blocks.contains(header.parent_hash());
+				let is_invalid = invalid_blocks.contains(block.header.parent_hash());
 				if is_invalid {
 					invalid_blocks.insert(hash);
 					continue;
 				}
 
+				let block_bytes = std::mem::take(&mut block.bytes);
 				match self.check_and_lock_block(block, client) {
```

Comment:
Hmm, this will `check_and_lock_block` with empty `block bytes`.

Then we must ensure that it and all functions it calls don't use `block bytes`, have you done it? 

Well, I propose to create another type `PreverifiedBlockWithBytes` or something. 
If that's not possible add big fat note which explains that `block bytes is empty`.

Reply:
> Then we must ensure that it and all the functions it calls don't use `block bytes`, have you done it?

I have yes. It is not used lower down in the call graph. I'll add a big comment about it.